src/core/i18n/utils.js

```diff
@@ -173,3 +175,26 @@ export function getLanguage({ lang, acceptLanguage } = {}) {
   // - normalization e.g: en-us -> en-US.
   return { lang: sanitizeLanguage(userLang), isLangFromHeader };
 }
+
+// moment uses locales like "en-gb" whereas we use "en_GB".
+export function makeMomentLocale(locale) {
+  return locale.replace('_', '-').toLowerCase();
+}
+
+// Create an i18n object with a translated moment object available we can
+// use for translated dates across the app.
+export function makeI18n(i18nData, _Jed = Jed) {
+  const i18n = new _Jed(i18nData);
+
+  // This adds the correct moment locale for the active locale so we can get
+  // localised dates, times, etc.
+  if (i18n.options && typeof i18n.options._momentDefineLocale === 'function') {
+    i18n.options._momentDefineLocale();
+    moment.locale(makeMomentLocale(i18n.options.locale_data.messages[''].lang));
```

Comment:
I tried that, but frustratingly the docs don't seem to point out that the `moment()` returned by `moment()` is a `moment` object with `new Date()` set as the time. There isn't a way to make new moments like `moment(addon.last_updated)`. And I don't see an API to reset the date, so this doesn't actually work :-(